com.github.FontManager.FontManager.yaml

```diff
@@ -0,0 +1,91 @@
+app-id: com.github.FontManager.FontManager
```

Comment:
This should be `io.github.` see the appid rules in checklist. The last part should be `font-manager` to match the repo name.

Reply:
I don't know how upstream chose the `com.github` id... they don't obviously own `github.com`

Reply:
Discussed upsteam. They want to carry this app Id and they maintain the metadata themselves:
https://github.com/FontManager/font-manager/issues/286

Reply:
What you want me to do? `org.gnome` also isn't ideal...

Reply:
Unless the id is fixed this cannot be accepted. New apps cannot have a com.github id. I suggest you contact upstream

Reply:
Or otherwise patch it to `io.github.fontManager.font-manager` if upstream is unwilling to fix it.

Reply:
Distributions like Debian Sid and Arch allow `com.github.FontManager.FontManager` so I doubt that Upsteam will care that much.

Perfect is the enemy of good. I'm afraid that it will take an unreasonable amount of time for me to patch the App-id. In which case, the package will stick with `org.gnome.FontManager`. 

Would this be worth an exception?

Reply:
No this breaks future verification and the reverse DNS check, with the current ID we calculate the domain to be fontmanager.github.com which does not exist.

Reply:
@JerryCasiano can you change the ID upstream to `io.github`. An ID should not be chosen on how it looks, it carries meaning that you control the domain. Otherwise anyone could claim a reversed domain as their ID.

This standard has been present for the past 20+ years in Linux.

See https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names

> For instance, if the owner of example.com is developing a D-Bus API for a music player, they might define interfaces called com.example.MusicPlayer1, com.example.MusicPlayer1.Track and com.example.MusicPlayer1.Seekable. 

https://www.freedesktop.org/software/appstream/docs/chap-Metadata.html#tag-id-generic

> Ownership of {vendor}.{tld} in the domain name system guarantees uniqueness of IDs. 

You were linked to both Flatpak and Flathub documentation for this in https://github.com/FontManager/font-manager/issues/286

Reply:
> You were linked to both Flatpak and Flathub documentation for this in FontManager/font-manager#286 (comment)

Interesting tone, almost feel like a child being scolded.

That said I'm still happy to review a pull request from you.

Reply:
Just to clarify, there won't be an exception for the app-id.

Reply:
> Interesting tone, almost feel like a child being scolded.

No one is scolding you. But this is the second time the app has the wrong ID when you were linked to the docs that clearly says not to use `com.github`. This is just frustrating for everyone.

Reply:
> No one is scolding you. But this is the second time the app has the wrong ID when you were linked to the docs that clearly says not to use com.github. This is just frustrating for everyone.

Second time?

I'm not sure what you're on about. 

I've changed the ID exactly once and it was changed to an ID which is clearly in line with one suggested as acceptable in the very thread you referenced, apparently you missed that part or simply chose to ignore it.


In any case, this is obviously not headed in a productive direction so if someone decides to make the necessary changes, great, I'll be happy to merge before the next point release otherwise oh well.

Unsubscribing from this noise.

Reply:
That suggestion was wrong the docs clearly said that. And the first id was `org.gnome` per https://github.com/FontManager/font-manager/issues/319 which was also a domain you didn't own.

Anyway not my problem, but this won't get into Flathub unless the ID is fixed either upstream or through a Flathub specific patch.

Reply:
@JerryCasiano 

> Interesting tone, almost feel like a child being scolded.

I'm sorry that you're being dragged into this. I maintain applications with creative App-Ids like `com.github.Flacon` and `io.github.ezQuake` and `org.gnome.gitlab.Cowsay`, without previous problems. Sadly, at some point in the past 4 years the policy and atmosphere has changed drastically.

I think we've done enough for now, and I'll gladly re-evaluate this issue once you've signed off on a merge request with the appropriate changes.

Reply:
> com.github.Flacon and io.github.ezQuake

I can tell you that I am 100% sure these two app-id wouldn't be accepted. They can't even be verified.

Reply:
> > com.github.Flacon and io.github.ezQuake
> 
> I can tell you that I am 100% sure these two app-id wouldn't be accepted. They can't even be verified.

We're trying to make the world (or at least, Flathub) a better place one small step at the time. Back when I submitted those applications, the rules were less strict. Importantly though, we all collaborated under the assumption that functional applications we're better then no applications. 

My point is not that those applications would now be allowed, but that there was a time when Flathub was more pragmatic. Perfect is the enemy of good.

Reply:
Those were wrong then, it's still wrong now. The only difference is that it wasn't enforced.

Reply:
Using a `or.gnome` ID made the GNOME foundation contact the project asking to change, using `com.github` makes it vulnerable to Github/MS down the line.